cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/AbstractInMemoryProgramRunner.java

```diff
@@ -87,7 +86,7 @@ protected final ProgramController startAll(Program program, ProgramOptions optio
         components.put(program.getName(), instanceId, controller);
       }
 
-      return new InMemoryProgramController(components, program, runId, options, programStateWriter);
+      return new InMemoryProgramController(components, program, options, runId);
```

Comment:
Yes there is - InMemoryProgramController extends the AbstractProgramController, which requires a ProgramId and RunId (i.e ProgramRunId)